common/src/main/java/com/wynntils/models/raid/raids/RaidKind.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2025.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.raid.raids;
+
+import com.wynntils.core.text.StyledText;
+import java.util.Map;
+
+public abstract class RaidKind {
+    private String raidName;
+    private String abbreviation;
+    private StyledText entryTitle;
+    private Map<Integer, Map<String, String>> challengeNames;
+    private int challengeCount;
+    private int bossCount;
+    private Map<String, Map<Integer, String>> majorIdBuffs;
+
+    protected RaidKind(
+            String raidName,
+            String abbreviation,
+            StyledText entryTitle,
+            Map<Integer, Map<String, String>> challengeNames,
+            int challengeCount,
+            int bossCount,
+            Map<String, Map<Integer, String>> majorIdBuffs) {
+        this.raidName = raidName;
+        this.abbreviation = abbreviation;
+        this.entryTitle = entryTitle;
+        this.challengeNames = challengeNames;
+        this.challengeCount = challengeCount;
+        this.bossCount = bossCount;
+        this.majorIdBuffs = majorIdBuffs;
+    }
```

Comment:
I'd create unmodifiable views of the maps set in the constructor, just in case. (Or alternatively, in the getters for them).